Frontend/ui-customer/src/Screens/NammaSafetyFlow/SafetySettingsScreen/View.purs

```diff
@@ -0,0 +1,432 @@
+                [ width MATCH_PARENT
+                , height WRAP_CONTENT
+                , orientation VERTICAL
+                , gravity LEFT
+                , padding $ Padding 16 16 16 16
+                ]
+                [ textView
+                    $ [ text $ getString EMERGENCY_ACTIONS
+                      , color Color.black900
+                      ]
+                    <> FontStyle.subHeading1 TypoGraphy
+                , textView
+                    $ [ text $ getString WHEN_YOU_START_EMERGENCY_SOS
+                      , color Color.black700
+                      ]
+                    <> FontStyle.body3 TypoGraphy
+                ]
+            , toggleSwitchViewLayout (ToggleSwitch SetDefaultEmergencyContacts) state.data.shareToEmergencyContacts (getString EMERGENCY_SHARING_WITH_CONTACTS) push true 16
+            , linearLayout
+                [ width MATCH_PARENT
+                , height WRAP_CONTENT
+                , orientation HORIZONTAL
+                , gravity CENTER_VERTICAL
+                , padding $ Padding 16 8 16 0
+                ]
+                [ textView
+                    $ [ text $ getString SHARING_WITH
+                      , color Color.black700
+                      , margin $ MarginRight 8
+                      , gravity CENTER
+                      , visibility $ boolToVisibility $ not $ null state.data.contactsList
+                      ]
+                    <> FontStyle.body3 TypoGraphy
+                , linearLayout
+                    [ height WRAP_CONTENT
+                    , width WRAP_CONTENT
+                    ]
+                    (mapWithIndex (\index item -> ContactCircle.view (ContactCircle.getContactConfig item index false) (push <<< ContactAction)) state.data.contactsList)
+                , textView
+                    $ [ text $ getString if null state.data.contactsList then ADD_CONTACTS else EDIT
+                      , color Color.blue900
+                      , margin $ MarginLeft 8
+                      , gravity CENTER
+                      , onClick push $ const $ EditEmergencyContacts
+                      ]
+                    <> FontStyle.body2 TypoGraphy
+                ]
+            , separatorView Color.lightGreyShade $ Margin 16 16 16 16
+            , toggleSwitchViewLayout (ToggleSwitch SetNightTimeSafetyAlert) state.data.nightSafetyChecks (getString NIGHT_TIME_SAFETY_CHECKS) push true 16
+            , separatorView Color.lightGreyShade $ Margin 16 16 16 16
+            , toggleSwitchViewLayout (ToggleSwitch SetShareTripWithContacts) state.data.shareTripWithEmergencyContacts (getString RIDE_SHARE_AFTER_SIX_PM) push true 16
+            , textView
+                $ [ text $ getString $ WHO_CAN_TRACK_YOUR_RIDE "WHO_CAN_TRACK_YOUR_RIDE"
+                  , color Color.black700
+                  , margin $ Margin 16 16 16 16
+                  , visibility $ boolToVisibility $ not $ null state.data.contactsList
```

Comment:
Can we store this in var `null state.data.contactsList` as it is being used multiple times